sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIService.scala

```diff
@@ -70,6 +70,77 @@ private[hive] class SparkSQLCLIService(hiveServer: HiveServer2, hiveContext: Hiv
       case _ => super.getInfo(sessionHandle, getInfoType)
     }
   }
+
+  private def withMetadataHive[T](sessionHandle: SessionHandle)(f: => T): T = {
+    val sessionConf = sparkSqlSessionManager.getSession(sessionHandle).getHiveConf
```

Comment:
In our own session manager, do we maintain a map from session handle to hive conf for metadata hive? Right now, the HiveSession associated with a session handle stores hive conf of the execution hive. 

Also, how does set command work?